broker.go

```diff
@@ -441,8 +475,9 @@ func (b *Broker) responseReceiver() {
 			continue
 		}
 
-		_, err = io.ReadFull(b.conn, header)
+		bytesReadHeader, err := io.ReadFull(b.conn, header)
 		if err != nil {
+			b.updateIncomingCommunicationMetrics(bytesReadHeader)
```

Comment:
I forgot that this code is running inside a `for` loop so using `defer` would require wrapping the `for` code block inside an function, not sure if it makes it easier to read.